network-formula/network/wicked/files/usr/local/sbin/saltsafe_if

```diff
@@ -0,0 +1,286 @@
+			result="changed=yes comment=\"$comment1 up\""
+		fi
+	elif [ "$call" == 'ifdown' ]
+	then
+		if ifstatus "$interface" -o quiet
+		then
+			result="changed=yes comment=\"$comment1 down\""
+		else
+			result="changed=no comment=\"$comment2 down\""
+		fi
+	fi
+}
+
+run_cycle() {
+	if run
+	then
+		if [ "$call" == 'ifdown' ]
+		then
+			log 0 "Brought down interface $interface."
+			result="changed=yes comment=\"Brought down $interface.\""
+			quit 0
+		fi
+		if check
+		then
+			if [ "$rollback" == 'yes' ]
+			then
+				if [ "$call" == 'ifroutes' ]
+				then
+					log 1 'Routing configuration rollback successful.'
+					result='changed=yes comment="Routing configuration reverted."'
+				else
+					log 1 'Interface configuration rollback successful.'
+					result='changed=yes comment="Interface configuration reverted."'
+				fi
+			else
+				log 0 'Operation and validation successful.'
+				result='changed=yes comment="Operation and validation successful."'
+				backup
+			fi
+			quit 0
+		else
+			if [ "$call" == 'ifroutes' ]
+			then
+				log 1 'Reloaded service, but validation failed.'
+				result='changed=yes comment="New routing configuration applied but failed."'
+			else
+				log 1 "Brought up $interface, but validation failed."
+				result="changed=yes comment=\"New configuration for interface $interface applied but failed.\""
+			fi
+			if [ "$rollback" = 'yes' ]
+			then
+				log 1 'Rollback was not successful. Giving up.'
+				if [ "$call" == 'ifroutes' ]
+				then
+					result='changed=yes comment="Failed to revert routing configuration."'
+				else
+					result='changed=yes comment="Failed to revert interface configuration."'
+				fi
+				quit 1
+			fi
+		fi
+	else
+		result='changed=yes comment="Execution failed."'
+		return "$?"
+	fi
+}
+
+if [ "$call" == 'ifroutes' ]
+then
+	filename='routes'
+	if [ "$extra" == 'test' ]
+	then
+		routes="${1?Cannot test without routes}"
+	fi
+else
+	interface="${1?Cannot operate without an interface}"
+	filename="ifcfg-$interface"
+
+	if ! command -v "$call" >/dev/null
+	then
+		fail "Unable to locate $call."
+	fi
+
+	if [ ! -f "$base_backup/ifcfg-$interface" ]
+	then
+		if [ "$extra" != 'test' ]
+		then
+			#fail "Missing backup configuration for interface $interface, refusing to operate."
+			backup
+		fi
+	fi
+fi
+
+file="$base/$filename"
+file_backup="$base_backup/$filename"
+
+# Get IP addresses of the Salt minion and master
+read minion_ip master_ip < <(ss -HntA tcp dst :4505 | awk '{ split($4, con_out, ":"); split($5, con_in, ":"); print con_out[1] " " con_in[1] }')
+
+if [ -z "$minion_ip" -o -z "$master_ip" ]
+then
+	fail 'Unable to determine Salt connection, refusing to operate.'
+fi
+
+# Get network interface the minion is using to connect to the master
+out_interface="$(ip -br a sh | awk -v ip=$minion_ip '$0 ~ ip { print $1 }')"
+
+danger=no
+rollback=no
+
+if [ "$call" == 'ifroutes' ]
+then
+	if [ "$(ip -ts r g $master_ip | awk -v ip=$master_ip '$0 ~ ip { print $2 }')" == 'via' ]
```

Comment:
For such "magic" lines, a comment explaining what they do would be nice ;-)

Reply:
Now I need to remember what this was about. ;-)

Reply:
Added!